### DISCUSSION TITLE
Background upload on iOS

### DISCUSSION THREAD

**Discussion by JanOstrowka** [ente-io/ente#855] [Enhancements]
Hi

I enjoy to using the Ente Photos app for securely storing and accessing my personal media library in the cloud. One feature that would greatly improve the app is the ability to upload photos and videos in the background.

I understand background uploads are tricky on iOS due to limits imposed by Apple. However, after seeing [the Insta360 app able to download content in the background](https://youtube.com/watch?v=Ql_c3G4aB9I&t=461), I'm curious if it may now be possible to implement background uploads too. Apps like Nextcloud and Proton Drive have struggled with this (especially Nextcloud), but perhaps the technical landscape has changed to make it viable now.

Implementing background uploads would allow uploads to continue when the app is closed or device is locked, ensuring media is backed up without constant active app usage draining battery. It would make cloud sync much smoother and effortless for users.

Please consider looking into adding background upload capabilities in a future iOS app update. Let me know if any testing would be helpful from a user perspective.

Thanks for your consideration!

**Comment by vishnukvmd** [comment]
Thanks for sharing the bit about Insta360, very interesting! However, there are currently no documented APIs for replicating this behavior, so it must be a clever hack.

Unfortunately we're swamped with work related to running on-device ML right now, once we've wrapped some of that up, we'll investigate this!

**Reply by vishnukvmd** [reply]
This seems promising: https://pub.dev/packages/background_downloader

**Reply by mkdirmushroom** [reply]
Is there any progress on this task at the moment?
